src/frontend/src/functions/forms.tsx

```diff
@@ -96,7 +187,28 @@ export function openModalApiForm(props: ApiFormProps) {
     return;
   }
 
-  let url = constructFormUrl(props);
+  // Generate a random modal ID for controller
+  let modalId: string =
+    `modal-${props.title}-${props.url}-${props.method}` + generateUniqueId();
+
+  props.actions = [
+    ...(props.actions || []),
+    {
+      text: props.cancelText ?? t`Cancel`,
+      color: props.cancelColor ?? 'blue',
+      onClick: () => {
+        modals.close(modalId);
+      }
+    }
+  ];
+
+  const oldFormSuccess = props.onFormSuccess;
+  props.onFormSuccess = (data) => {
+    oldFormSuccess?.(data);
+    modals.close(modalId);
+  };
```

Comment:
@SchrodingersGat To continue [this](https://github.com/inventree/InvenTree/pull/5947#discussion_r1399160566) conversation here. This is just a name, because I need to wrap the `onFormSuccess' function so that I can do something else and also call the callback provided in props. If I wouldn't save this in a variable, and just call it from inside, I would have an endless recursion.
![image](https://github.com/inventree/InvenTree/assets/76838159/ae9cffdc-1e63-4ecb-8f64-238fb6383944)